cmake/projects/OpenSSL/schemes/url_sha1_openssl_macos.cmake.in

```diff
@@ -0,0 +1,220 @@
+else()
+  set(shared_library_version_extension "1.1")
+endif()
+
+if(BUILD_SHARED_LIBS)
+  set(libsuffix ".${shared_library_version_extension}.dylib")
+  set(shared_flag shared)
+else()
+  set(libsuffix ".a")
+  set(shared_flag no-shared)
+endif()
+
+hunter_unsetvar(ssl_libraries)
+hunter_unsetvar(crypto_libraries)
+foreach(arch ${configure_architectures})
+  list(APPEND ssl_libraries "@HUNTER_PACKAGE_SOURCE_DIR@/${arch}/libssl${libsuffix}")
+  list(APPEND crypto_libraries "@HUNTER_PACKAGE_SOURCE_DIR@/${arch}/libcrypto${libsuffix}")
+endforeach()
+
+# Skip position independent code flags for macOS
+hunter_dump_cmake_flags(SKIP_INCLUDES SKIP_PIC)
+# -> CMAKE_CXX_FLAGS
+# -> CMAKE_C_FLAGS
+
+# Pass C compiler through
+set(configure_command
+    MACHINE=${CMAKE_SYSTEM_PROCESSOR}
+    CC=${CMAKE_C_COMPILER}
+    ${configure_command})
+
+# Pass C flags through
+set(configure_opts ${CMAKE_C_FLAGS})
+
+# Add clear-all to commands
+set(configure_command . "@HUNTER_GLOBAL_SCRIPT_DIR@/clear-all.sh" && "${configure_command}")
+set(build_command . "@HUNTER_GLOBAL_SCRIPT_DIR@/clear-all.sh" && make)
+
+# Configuration options
+set(configure_opts ${configure_opts} threads ${shared_flag})
+
+hunter_status_debug("Configure command: ${configure_command}")
+hunter_status_debug("Configure architectures: ${configure_architectures}")
+hunter_status_debug("Configure options: ${configure_opts}")
+
+# fuse all libraries with 'lipo'
+ExternalProject_Add(
+    "@HUNTER_EP_NAME@-macos_universal"
+    DOWNLOAD_COMMAND
+    "@CMAKE_COMMAND@" -E echo "Dummy download command"
+    DOWNLOAD_DIR
+    "@HUNTER_PACKAGE_DOWNLOAD_DIR@"
+    TLS_VERIFY
+    "@HUNTER_TLS_VERIFY@"
+    SOURCE_DIR
+    "@HUNTER_PACKAGE_SOURCE_DIR@/macos_universal"
+    INSTALL_DIR
+    "@HUNTER_PACKAGE_INSTALL_PREFIX@"
+        # not used, just avoid creating Install/<name> empty directory
+    CONFIGURE_COMMAND
+    "@CMAKE_COMMAND@" -E echo "Dummy configure command"
+    BUILD_COMMAND
+    "@CMAKE_COMMAND@" -E echo "Dummy build command"
+    INSTALL_COMMAND
+    lipo
+    -create
+    ${ssl_libraries}
+    -o
+    "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/libssl${libsuffix}"
+    COMMAND
+    lipo
+    -create
+    ${crypto_libraries}
+    -o
+    "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/libcrypto${libsuffix}"
+)
+
+if(BUILD_SHARED_LIBS)
+  # Fix the dylib loading paths
+  ExternalProject_Add(
+      "@HUNTER_EP_NAME@-fix_install_name"
+      DOWNLOAD_COMMAND
+      "@CMAKE_COMMAND@" -E echo "Dummy download command"
+      DOWNLOAD_DIR
+      "@HUNTER_PACKAGE_DOWNLOAD_DIR@"
+      TLS_VERIFY
+      "@HUNTER_TLS_VERIFY@"
+      SOURCE_DIR
+      "@HUNTER_PACKAGE_SOURCE_DIR@/macos_universal"
+      INSTALL_DIR
+      "@HUNTER_PACKAGE_INSTALL_PREFIX@"
+          # not used, just avoid creating Install/<name> empty directory
+      CONFIGURE_COMMAND
+      "@CMAKE_COMMAND@" -E echo "Dummy configure command"
+      BUILD_COMMAND
+      "@CMAKE_COMMAND@" -E echo "Dummy build command"
+      INSTALL_COMMAND
```

Comment:
>  had to use a command to updated the shared libraries I kept the second ExternalProject_Add

You can append it to `"@HUNTER_EP_NAME@-macos_universal"`:

```cmake
if(BUILD_SHARED_LIBS)
  set(
      extra_install_commands
      COMMAND
      install_name_tool
      -change
      # ...
  )
else()
  set(extra_install_commands "")
endif()

ExternalProject_Add(
    "@HUNTER_EP_NAME@-macos_universal"
    # ...
    INSTALL
    lipo
    -create
    # ...
    ${extra_install_commands}
)
```

Reply:
Ah, I didn't know that was possible. Thanks.